spec/support/examples_group/shared_examples_for_provisioning_state_machine.rb

```diff
@@ -44,3 +44,32 @@
     end
   end
 end
+
+shared_examples_for "End-to-end State Machine Run" do
+  it "Completes Successfully" do
+    expect(task).to be_kind_of(ManageIQ::Providers::Redhat::InfraManager::Provision)
+    task.options[:vm_target_name] = options[:vm_target_name] # HACK: Automate usually does this
+
+    @queue           = []
+    @called_states   = []
+    @signaled_states = []
+
+    allow(task).to receive(:signal).and_wrap_original do |_method, *args|
+      @called_state = args.first
+      expect(expected_states_with_counts.keys).to include(@called_state)
+      @signaled_states << @called_state
+      @queue |= [@called_state]
+    end
+
+    task.run_provision # Get the state machine rolling
+
+    loop { dequeue_method || break }
```

Comment:
This simulates the queued states being dequeued and executed.  Break when nothing is left on the queue.